src/eth/rpc/rpc_server.rs

```diff
@@ -157,7 +159,8 @@ fn register_methods(mut module: RpcModule<RpcContext>, env: Environment) -> anyh
 // Debug
 async fn debug_set_head(params: Params<'_>, ctx: Arc<RpcContext>) -> anyhow::Result<JsonValue, RpcError> {
     let (_, number) = next_rpc_param::<BlockNumber>(params.sequence())?;
-    ctx.storage.reset(number).await?;
+    TemporaryStorage::reset(&*ctx.storage).await?;
```

Comment:
I think that maybe it would make more sense for StratusStorage to implement reset, and both storages reset functions be called from there. I think that the logic of interacting with permanent and temporary storages should be isolated in StratusStorage if possible.